wikimedia/wikipedia-ios
# Update top read widget to be language variant aware

**Phabricator:** https://phabricator.wikimedia.org/T277697 (this PR is a stepping stone towards addressing this task)

### Notes
Previously, the top read widget would get into states where the title and description would be a. displayed in contrasting variants from one another and b. inconsistent with what's displayed in the Explore feed in the app. For example, if you were using `Serbian (Latin)` as your primary language, you might see a `Serbian (Latin)` description but a `Serbian (Cyrillic)` title.

While it's certainly possible you may still see mixed character sets (for backend content reasons), this PR seeks to reach some sense of parity between the top read widget's displayed titles and descriptions and the in-app Explore feed's top read titles and descriptions.

- Updates the top read widget to use variant aware URLs
- Populates the widget elements' title and description from the fetched `WMFArticle` instead of the `WMFFeedTopReadArticlePreview`

### Test Steps
I don't have great test steps here because getting the widgets to reload as desired has always been problematic, but generally:

1. Make your preferred language one with many variants
2. Observe the top read widget title and description, ensure they match the in-app Explore feed's top read elements (specifically, that the title and description are using the same variant as the Explore feed's top read group)
3. Change to a different variant, repeat step 2.

Sometimes you'll encounter cases of the widgets not reloading, in which case sometimes deleting and re-adding the widget helps. The intent of this PR is not so much to address the reloading problem, but to help address the consistency issues of the titles and descriptions as described in step 2.

![top_read](https://user-images.githubusercontent.com/5652327/112703391-fcd24300-8e53-11eb-8650-f962de311142.png)
